### PR TITLE
[Docs] Added the Diesel ORM for Rust in Drivers & ORMs Section

### PR DESCRIPTION
**Pull request by priyanshi-yb**
Added the Diesel ORM for Rust in Drivers & ORMs Section

### CODE REVIEW

**Comment by polarweasel** [comment]
@priyanshi-yb if this PR is ready for review, please take it out of draft mode and we'll get it reviewed, edited, and published.

File: docs/content/preview/drivers-orms/rust/_index.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    weight: 570
+isTocNested: true
+showAsideToc: true
+---
+
+Following are the recommended projects for implementing Rust Applications for YugabyteDB YSQL API.
+
+| Project | Type | Support Level |
+| :------ | :--- | :------------ |
+| [Diesel](diesel) | ORM | Full |
```
Comment (polarweasel) [inline docs/content/preview/drivers-orms/rust/_index.md:20]: Trailing slashes on links
```suggestion
| [Diesel](diesel/) | ORM | Full | 
```

File: docs/content/preview/drivers-orms/rust/diesel.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+YugabyteDB YSQL API has full compatibility with Diesel ORM for Data persistence in Rust applications. This page provides details for getting started with Diesel ORM for connecting to YugabyteDB.
+
+## Create a New Rust-Diesel Project 
+
+1. First, Ensure you have Rust (1.58.1) and Cargo(1.58.0) installed. If not installed. 
+2. Create a new project by using this:
+
+```shell
+$ cargo new --lib diesel_demo && cd diesel_demo
```
Comment (polarweasel) [inline docs/content/preview/drivers-orms/rust/diesel.md:25]: This command doesn't succeed on my macOS 12.3 machine WITH libpq installed. I believe there's at least one environment variable that we need to be setting here.

**Comment by polarweasel** [comment]
@priyanshi-yb please sign the CLA...

Also please see my comment about where the build is failing for me.

**Comment by priyanshi-yb** [comment]
> @priyanshi-yb please sign the CLA...

I have signed it.

> Also please see my comment about where the build is failing for me.

Can you please let me know the error you are getting ?

**Review by polarweasel — APPROVED** [review]

**Pull request merged**